lightning/src/ln/channelmanager.rs

```diff
@@ -3081,6 +3153,24 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 														},
 														_ => panic!(),
 													}
+												} else if fake_scid::is_valid_intercept(&self.fake_scid_rand_bytes, short_chan_id) {
+													let intercept_id = InterceptId(Sha256::hash(&incoming_shared_secret).into_inner());
+													let mut pending_intercepts = self.pending_intercepted_payments.lock().unwrap();
+													match pending_intercepts.entry(intercept_id) {
+														hash_map::Entry::Vacant(entry) => {
+															entry.insert(forward_info);
+															new_events.push(events::Event::PaymentIntercepted {
+																short_channel_id: short_chan_id,
+																payment_hash,
+																inbound_amount_msats: 0,
```

Comment:
I'm not sure best way to get this or if it's really that important to surface it? Does it need to be piped through from earlier steps?

Reply:
yeah that makes sense.  we can fail it safely because if it's missing that attribute it means it was created on a version that didn't support interception, right?